avocado/utils/dmesg.py

```diff
@@ -84,7 +85,7 @@ def collect_errors_dmesg(patterns):
     return error
 
 
-def collect_errors_by_level(output_file=None, level_check=5):
+def collect_errors_by_level(output_file=None, level_check=5, skip_error=None):
```

Comment:
This method is getting "too complex" and doing multiple things. My suggestion is to keep using as it was and use your new method to filter:

```python
messages = collect_erros_by_level()
filtered = skip_dmesg_error(messages, filter)
```

Reply:
this is fine but collect_erros_by_level() will not return the dmesg messages it will raise exception if any messages found. if i will do this approach i need to do some changes in collect_erros_by_level(). so i have included directly in that function.